### DISCUSSION TITLE
章节标题行间距及段前段后设置问题

### DISCUSSION THREAD

**Discussion by Komari-Koshigaya** [mengchaoheng/SCUT_thesis#19] [Q&A]
> 撰写规范要求 “章、节、条三级标题为单倍行距，段前、段后各设为0.5行（即前后各空0.5行）”

最近有点时间，挨个对了一下格式，想确认一下cls文件的设置是否满足以上要求。仔细看了一下，有些许疑惑：

```cls
% line 367
\titleformat{\chapter}[hang]{\textchapterfont}{第\xCJKnumber{\thechapter}章}{1em}{}
\titleformat{\section}[hang]{\textsectionfont}{\thesection}{1em}{}
\titleformat{\subsection}[hang]{\textsubsectionfont}{\thesubsection}{1em}{}
\titleformat{\subsubsection}[hang]{\textsubsubsectionfont}{\thesubsubsection}{1ex}{}% 小四黑体居左
```
1. 上述代码是设置单倍行距吗？ 对应的是`1em`?

```cls
% line 374
\titlespacing{\chapter}{0pt}{-20pt}{1pt} % 缩短章标题上间距，revised by Quansheng Guan
\titlespacing{\section}{0pt}{1pt}{1pt}  % 标题间距
\titlespacing{\subsection}{0pt}{4pt}{0pt}
\titlespacing{\subsubsection}{0pt}{6pt}{6pt}
```
2. 上述代码是在设置段前段后0.5行吗？ 1pt、4pt、6pt是如何计算出来的？
3. 为什么`\titlespacing{\subsection}{0pt}{4pt}{0pt} `最后一个是0pt，而不是4pt？

**Comment by mengchaoheng** [comment]
1. 那个好像不是，你可以测试一下。最好参考最新的vscode编译，时间缩短至原来的一半。
2. 这部分都是祖传代码，当初时间有限并没有研究这部分。如果要深入研究可以看latex的书，最新的模版我上传了一些资料。我个人是鼓励大家积极探索，但很多同学表示只想使用，并不想成为latex专家，所以最开始的文档我做得很白话文（毕竟我也算小白）。
3. 同理，我也不知道🤣。我只知道有些数字例如行距是可以算出来的，然后有些甚至就手调出来，因为行距有个比例因子的设置，什么”某系数的1.2倍再乘某系数因子“，看得我头痛，暂时没有时间搞透。

如果我有时间做这部分的话我先修改一下值观察效果，然后百度某指令的含义，然后结合latex说明文档/书籍进行进一步了解。最后修改该指令或者完全用新的代码重写。需要用到某些宏包的时候再看看包的说明文档。

最后，我会全力配合进行祖传代码的改进。当初我只是更改部分，源代码都留在注释中。我也发现了很多不清楚的代码以及多余的设置。但无力进行大改。原模版以及我修改后的模版已经经历很多届同学的实践证明可用，这也是我赖于重新整理原模版的原因。

**Comment by mengchaoheng** [comment]
我想到一个思路，就是保留参考文献用biber，论文前面的页码和原来一样用word转pdf覆盖。正文图片表格这些也用宏包，公式就是普通的公式，正文其他文字一律重写cls文件。总之就是能用包的就用包，各大纲级别的字体、字号、段落格式就内置命令进行控制，命令写到cls文件。这样模版会很清晰。然而问题是，虽然代码不多但是工程其实挺庞大的，需要对模版的设计有一定的了解，需要看一些教程，还需要参考一下其他学校模版案例。但是准毕业生往往没时间，毕业后更难有连续的时间，且不说兴趣了。目前活跃的贡献者就你们两位，还是需要更多的人加入。
@Komari-Koshigaya @zhuohoudeputao

**Reply by Komari-Koshigaya** [reply]
> 我想到一个思路，就是保留参考文献用biber，论文前面的页码和原来一样用word转pdf覆盖。正文图片表格这些也用宏包，公式就是普通的公式，正文其他文字一律重写cls文件。总之就是能用包的就用包，各大纲级别的字体、字号、段落格式就内置命令进行控制，命令写到cls文件。这样模版会很清晰。然而问题是，虽然代码不多但是工程其实挺庞大的，需要对模版的设计有一定的了解，需要看一些教程，还需要参考一下其他学校模版案例。但是准毕业生往往没时间，毕业后更难有连续的时间，且不说兴趣了。目前活跃的贡献者就你们两位，还是需要更多的人加入。 @Komari-Koshigaya @zhuohoudeputao

现在不就是这么干的吗？  你是说把scutthesis..tex导言区的那些控制也写挪到cls文件里去

**Reply by mengchaoheng** [reply]
> > 我想到一个思路，就是保留参考文献用biber，论文前面的页码和原来一样用word转pdf覆盖。正文图片表格这些也用宏包，公式就是普通的公式，正文其他文字一律重写cls文件。总之就是能用包的就用包，各大纲级别的字体、字号、段落格式就内置命令进行控制，命令写到cls文件。这样模版会很清晰。然而问题是，虽然代码不多但是工程其实挺庞大的，需要对模版的设计有一定的了解，需要看一些教程，还需要参考一下其他学校模版案例。但是准毕业生往往没时间，毕业后更难有连续的时间，且不说兴趣了。目前活跃的贡献者就你们两位，还是需要更多的人加入。 @Komari-Koshigaya @zhuohoudeputao
> 
> 现在不就是这么干的吗？ 你是说把scutthesis..tex导言区的那些控制也写挪到cls文件里去

现在乱且有大量无用代码。

**Reply by mengchaoheng** [reply]
> 现在不就是这么干的吗？ 你是说把scutthesis..tex导言区的那些控制也写挪到cls文件里去

是的，有的格式控制还放在章节文件，确实有必要重写了。

**Comment by Komari-Koshigaya** [comment]
这个段落间距还是没搞懂，算了 先不管这个了
@mengchaoheng itemize和enumerate环境中item之间的间距 是不是有点大啊？（如下图）
![image](https://user-images.githubusercontent.com/39209375/164141674-230cf7e7-ab0d-4a5b-a2ef-388560b0abdc.png)

你有没有什么思路啊，我看网上用的`setitemize`和`setlist`都没用，甚至报错

**Reply by mengchaoheng** [reply]
这样子挺美观啊，还有论文中很少用到这个环境所以。。。以前是百度来的命令。

**Reply by Komari-Koshigaya** [reply]
> 这样子挺美观啊，还有论文中很少用到这个环境所以。。。以前是百度来的命令。

但是这样是不是不符合 正文1.5倍行距的要求

**Reply by Komari-Koshigaya** [reply]
> 这样子挺美观啊，还有论文中很少用到这个环境所以。。。以前是百度来的命令。

你论文中不用分点吗？ 还是说你自己手动给序号

**Reply by mengchaoheng** [reply]
是的，手动给序号。

**Reply by mengchaoheng** [reply]
我也忘了，我实验室的有人手动。但我的[论文](https://github.com/mengchaoheng/SCUTthesis-mengchaoheng)用了什么我也忘了。你可以点超链接进去看看。

**Reply by Komari-Koshigaya** [reply]
> 我也忘了，我实验室的有人手动。但我的[论文](https://github.com/mengchaoheng/SCUTthesis-mengchaoheng)用了什么我也忘了。你可以点超链接进去看看。

你论文中也是用的enumerate环境 

我找到了一个可用的设置 https://github.com/NewFuture/NKThesis/blob/73fada49a5fa8b253d588e597e24327c765e0e31/NKThesis.sty#L721

在自己的论文中试了一下可以用，你觉得模板需要改吗

**Reply by mengchaoheng** [reply]
如果测试没问题，不影响其他格式就可以加上。模版您可以随意更改，只要有利于大家写论文就可以。欢迎推pr过来。我可能到今年下半年回学校读博，也许到时候才有空对这个模版进行大改了。现在我主要在GitHub上搞机器人领域的开发，根本没空看模版仓库。这几天一直在各种开源机器人库中徘徊乱逛🤣
